File: CSharp/PaymentsService/Infrastructure/Logging/ConsoleLogger.cs
using PaymentsService.Domain.Logging;

namespace PaymentsService.Infrastructure.Logging;

public class ConsoleLogger : IBaseLogger
{
    public void Log(string message) => Console.WriteLine(message);

    public void Log<T>(T @object) => Log(Newtonsoft.Json.JsonConvert.SerializeObject(@object));

    public void Log<T>(string message, T @object)
    {
        Log(message);
        Log(@object);
    }
}

File: Python/PaymentsService/Infrastructure/Logging/ConsoleLogger.py
from Domain.Logging.IBaseLogger import IBaseLogger


class ConsoleLogger(IBaseLogger):
    def log(self, message):
        print(message)

    def log_object(self, obj):
        print(obj)

    def log_message_and_object(self, message, obj):
        print(f"{message}: {obj}")